src/opm/parser/eclipse/EclipseState/checkDeck.cpp

```diff
@@ -48,6 +48,49 @@ bool checkDeck( const Deck& deck, const Parser& parser, size_t enabledChecks) {
         deckValid = deckValid && Section::checkSectionTopology(deck, parser, ensureKeywordSection);
     }
 
+    // ensure that only a single unit system is used by the deck
+    int numUnitSystems = 0;
+    if (deck.hasKeyword("FIELD"))
+        ++ numUnitSystems;
+    if (deck.hasKeyword("METRIC"))
+        ++ numUnitSystems;
+    if (deck.hasKeyword("LAB"))
+        ++ numUnitSystems;
+    if (deck.hasKeyword("PVT-M"))
+        ++ numUnitSystems;
+
+    if (numUnitSystems != 1) {
+        std::string msg("Exactly one unit system must be specified by the deck (is: " + std::to_string(numUnitSystems) + ").");
```

Comment:
It is perfectly OK to not specify any unit system - in that case `METRIC` is the default.

Reply:
fixed. thanks!